src/java.base/share/classes/java/lang/Class.java

```diff
@@ -3820,8 +3820,9 @@ public boolean isRecord() {
     // Fetches the factory for reflective objects
     @SuppressWarnings("removal")
     private static ReflectionFactory getReflectionFactory() {
+        ReflectionFactory reflectionFactory = Class.reflectionFactory;
         if (reflectionFactory == null) {
-            reflectionFactory =
+            return Class.reflectionFactory =
                 java.security.AccessController.doPrivileged
                     (new ReflectionFactory.GetReflectionFactoryAction());
         }
```

Comment:
p.s.  If using an early return, I'd have it return early in the case reflectionFactory was not-null.
then fall through to compute and assign the factory and return it.
```
        var factory = reflectionFactory;
        if (factory != null) {
            return factory;
        }
        return reflectionFactory =
            java.security.AccessController.doPrivileged
                (new ReflectionFactory.GetReflectionFactoryAction());
```